app/dashboard/views.py

```diff
@@ -776,3 +785,75 @@ def toolbox(request):
         'newsletter_headline': "Don't Miss New Tools!"
     }
     return TemplateResponse(request, 'toolbox.html', context)
+
+
+@csrf_exempt
+@ratelimit(key='ip', rate='5/m', method=ratelimit.UNSAFE, block=True)
+def redeem_coin(request, shortcode):
+    if request.body:
+        status = 'OK'
+
+        body_unicode = request.body.decode('utf-8')
+        body = json.loads(body_unicode)
+        address = body['address']
+
+        try:
+            coin = CoinRedemption.objects.get(shortcode=shortcode)
```

Comment:
using this code... whats to stop me from just using the same `shortcode` over and over again

Reply:
^ Agreed.  Maybe we should check for the existence  of a `CoinRedemptionRequest` for the `coin` before performing the transfer.

Reply:
^^Bummer. I've added that check for `GET` request and totally forgot about the similar issue for the API apart. Let me quickly do that. Thanks for pointing this out 👍

Reply:
@owocki @mbeacom -- This has been taken care of now